pandas/core/common.py

```diff
@@ -2246,6 +2254,42 @@ def _join_unicode(lines, sep=''):
                          for x in lines])
 
 
+def _justify(text, max_len, mode='right', strlen=len):
+    """
+    Perform ljust, center, rjust against string or list-like
+    Text width is calcurated using strlen func
+    """
+
+    if not is_list_like(text):
+        # only accept list as input to avoid repeated check of padding
+        raise ValueError('input must be list-like')
+
+    # format._strlen_func return strlen wrapping with functools.partial
+    if isinstance(strlen, partial):
```

Comment:
I think this check should be done higher up, with the strlen just passed in directly (e.g. based on the option in the global namespace)